src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/SeedTemporaryAccommodationPremisesTest.kt

```diff
@@ -0,0 +1,270 @@
+      it.level == "error" &&
+        it.message == "Error on row 1:" &&
+        it.throwable != null &&
+        it.throwable.message == "Probation Region Not Real Region does not exist"
+    }
+  }
+
+  @Test
+  fun `Attempting to create a Temporary Accommodation Premises with an invalid Local Authority Area logs an error`() {
+    val probationRegion = probationRegionEntityFactory.produceAndPersist {
+      withApArea(apAreaEntityFactory.produceAndPersist())
+    }
+
+    withCsv(
+      "invalid-local-authority-ta",
+      temporaryAccommodationPremisesSeedCsvRowsToCsv(
+        listOf(
+          TemporaryAccommodationPremisesSeedCsvRowFactory()
+            .withProbationRegion(probationRegion.name)
+            .withLocalAuthorityArea("Not Real Authority")
+            .produce()
+        )
+      )
+    )
+
+    seedService.seedData(SeedFileType.temporaryAccommodationPremises, "invalid-local-authority-ta")
+
+    assertThat(logEntries).anyMatch {
+      it.level == "error" &&
+        it.message == "Error on row 1:" &&
+        it.throwable != null &&
+        it.throwable.message == "Local Authority Area Not Real Authority does not exist"
+    }
+  }
+
+  @Test
+  fun `Creating a new Temporary Accommodation Premises persists correctly`() {
+    val probationRegion = probationRegionEntityFactory.produceAndPersist {
+      withApArea(apAreaEntityFactory.produceAndPersist())
+    }
+
+    val localAuthorityArea = localAuthorityEntityFactory.produceAndPersist()
+
+    val csvRow = TemporaryAccommodationPremisesSeedCsvRowFactory()
+      .withProbationRegion(probationRegion.name)
+      .withLocalAuthorityArea(localAuthorityArea.name)
+      .withCharacteristics(listOf("Park nearby", "Not suitable for arson offenders"))
+      .produce()
+
+    withCsv(
+      "new-ta-premises",
+      temporaryAccommodationPremisesSeedCsvRowsToCsv(
+        listOf(
+          csvRow
+        )
+      )
+    )
+
+    seedService.seedData(SeedFileType.temporaryAccommodationPremises, "new-ta-premises")
+
+    val persistedPremises = temporaryAccommodationPremisesRepository.findByName(csvRow.name)
+    assertThat(persistedPremises).isNotNull
+    assertThat(persistedPremises!!.addressLine1).isEqualTo(csvRow.addressLine1)
+    assertThat(persistedPremises.addressLine2).isEqualTo(csvRow.addressLine2)
+    assertThat(persistedPremises.town).isEqualTo(csvRow.town)
+    assertThat(persistedPremises.postcode).isEqualTo(csvRow.postcode)
+    assertThat(persistedPremises.probationRegion.name).isEqualTo(csvRow.probationRegion)
+    assertThat(persistedPremises.localAuthorityArea!!.name).isEqualTo(csvRow.localAuthorityArea)
+    assertThat(persistedPremises.pdu).isEqualTo(csvRow.pdu)
+    assertThat(persistedPremises.characteristics.map { it.name }).isEqualTo(csvRow.characteristics)
+    assertThat(persistedPremises.notes).isEqualTo(csvRow.notes)
+  }
+
+  @Test
+  fun `Updating an existing Temporary Accommodation Premises persists correctly`() {
```

Comment:
Good to see we're testing for idempotentcy 👍 My main concern with this type of process is what happens if an error is thrown half way through or we need to re-run this.

Reply:
The seeding process is set up so that if an error occurs within a row, the error is logged and the row is skipped so that following rows can be processed. Correcting the CSV file and rerunning the job should work fine, as I've done this multiple times during my manual testing.